backend/pkg/scheduler/scheduler.go

```diff
@@ -0,0 +1,166 @@
+	return &Scheduler{
+		stopChan:  make(chan struct{}),
+		scanChan:  scanChan,
+		dbHandler: dbHandler,
+	}
+}
+
+func (s *Scheduler) Init() {
+	// read last schedule scan config from db
+	sched, err := s.dbHandler.SchedulerTable().Get()
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			log.Info("No scheduler config")
+			return
+		}
+		log.Errorf("Failed to get scheduler table: %v", err)
+		return
+	}
+	scanConfig := models.RuntimeScheduleScanConfig{}
+	if err := json.Unmarshal([]byte(sched.Config), &scanConfig); err != nil {
+		log.Errorf("Failed to unmarshal scheduler config: %v", err)
+		return
+	}
+	startTime, err := time.Parse(time.RFC3339, sched.NextScanTime)
+	if err != nil {
+		log.Errorf("Failed to parse nextScanTime: %v. %v", sched.NextScanTime, err)
+		return
+	}
+
+	// start schedule scan
+	s.Schedule(&SchedulerParams{
+		Namespaces:                    scanConfig.Namespaces,
+		CisDockerBenchmarkScanEnabled: scanConfig.CisDockerBenchmarkScanEnabled,
+		Interval:                      time.Duration(sched.Interval),
+		StartTime:                     startTime,
+		SingleScan:                    scanConfig.ScanConfigType().ScheduleScanConfigType() == SingleScheduleScanConfig,
+	})
+}
+
+func (s *Scheduler) Schedule(params *SchedulerParams) {
+	// Clear
+	close(s.stopChan)
+	s.stopChan = make(chan struct{})
+
+	startsAt := getStartsAt(time.Now().UTC(), params.StartTime, params.Interval)
+
+	go s.spin(params, startsAt)
+}
+
+// get the next scan, that is after timeNow. if currentScanTime is already after timeNow, it will be return.
+func getNextScanTime(timeNow, currentScanTime time.Time, interval time.Duration) time.Time {
+	// if current scan time is before timeNow, jump to the next future scan time
+	if currentScanTime.Before(timeNow) {
+		// if scan time has passed in less then a second, start a scan now.
+		timePassed := timeNow.Sub(currentScanTime)
+		if timePassed < time.Second {
+			return timeNow
+		}
+		remainingInterval := timePassed % interval
+		if remainingInterval == 0 {
+			currentScanTime = timeNow
+		} else {
+			currentScanTime = timeNow.Add(interval - remainingInterval)
+		}
+	}
+	return currentScanTime
+}
+
+// get the time in Duration that the next scan should start at.
+func getStartsAt(timeNow time.Time, startTime time.Time, interval time.Duration) time.Duration {
+	nextScanTime := getNextScanTime(timeNow, startTime, interval)
+
+	startsAt := nextScanTime.Sub(timeNow)
+
+	return startsAt
+}
+
+func (s *Scheduler) spin(params *SchedulerParams, startsAt time.Duration) {
+	log.Debugf("Starting a new schedule scan. interval: %v, start time: %v, starts in: %v, namespaces: %v, cisDockerBenchmarkScanEnabled: %v",
+		params.Interval, params.StartTime, startsAt, params.Namespaces, params.CisDockerBenchmarkScanEnabled)
+	singleScan := params.SingleScan
+	interval := params.Interval
+
+	timer := time.NewTimer(startsAt)
+	select {
```

Comment:
missing `defer timer.Stop()`

Reply:
I dont think it is necessary to stop a timer.

Reply:
why not to stop the timer if you stopped due to stop channel?